versioned_docs/version-2.0.0/firebase.md

```diff
@@ -0,0 +1,51 @@
+---
+id: firebase
+title: Set up Firebase
+---
+
+Honeycomb comes with methods and configurations for easy [Firebase](https://firebase.google.com/) and [Cloud Firestore](https://firebase.google.com/docs/firestore) set up. Please use this section to configure [Firebase Hosting](https://firebase.google.com/docs/hosting) services and firestore cloud storage.
+
+## Getting Started
+
+Follow these steps to create a firebase project and link it with the current task.
+
+#### 1. Initializing firebase account and project
+- Create and login to a firebase account on [the Firebase website](https://firebase.google.com/).
+- Create a firebase project by clicking add project and enter a Project Name.
+
+#### 2. Linking firebase to task
+- Install [Firebase CLI](https://firebase.google.com/docs/cli) on your local computer.
+- Login to firebase using `firebase login` command in the terminal.
+- Navigate to the `.firebaserc` file home directory and edit the `"default"` field with the project name given in part 1.
+    ```
+    {
+        "projects": {
+            "default": "<your project name>"
+        }
+    }   
+    ``` 
+#### 3. Copying web app credentials.
+- Navigate to the [firebase console](https://console.firebase.google.com/) and select the project.
+- Click `Add App` and choose to create a wep app. Enter app name and click `Register App`.
+- Copy these fields in the firebase SDK to the corresponding variable in the `.env.firebase` file in the `env` folder:
+  ```
+    REACT_APP_apiKey=
+    REACT_APP_authDomain=
+    REACT_APP_projectId=
+    REACT_APP_storageBucket=
+    REACT_APP_messagingSenderId=
+    REACT_APP_appId=
+  ```
+Firebase is now set up!
+
+## Setting up firestore.
+
+Honeycomb includes a set of security rules to store participant data in firestore. Follow these steps to configure firestore
+
+- Navigate to the current project in the developer console and select `Firestore Database` from the sidebar.
+- Click `Create Database`, select `production mode` and choose the current location for the cloud storage bucket.
+- Navigate to the `Rules` tab once the database is created then copy the code that is in the `firestore.rules` file.
```

Comment:
While explaining how to update rules from the Firebase UI is good, I think it would be best that we suggest that the rules are pushed directly from the repo with `firebase deploy`  or `firebase deploy --only firestore:rules` no? That way the rules are more explicitly documented.
I'm also just realizing that the GitHub Actions that get generated only deploy hosting... meaning the rules don't get deployed? Should we provide guidance on how to add that to the actions? @mcmcgrath13 what's you view on this?

Reply:
I tend to manually deploy rules (either with `deploy` or copy/paste) as I often roll out backwards compatible firestore changes before the frontend code is ready so that I can do more thorough testing on the preview branches.  But sometimes this isn't viable/wise, so it depends.  We could maybe automate the deploy on merge to main, but sometimes they need to be updated earlier manually.